index.bs

```diff
@@ -1160,4298 +1160,4345 @@ access to that data in a {{Window}} global.
+   1. Let |serialized value| be |serialized key-value|[1].
 
-[=remote end definition=]
+   1. Let |deserialized value| be result of [=trying=] to [=deserialize local value=]
+      given |serialized value|, |realm| and |session|.
 
-<pre class="cddl remote-cddl">
+   1. Append [=CreateArrayFromList=](«|deserialized key|, |deserialized value|») to
+      |deserialized key-value list|.
 
-NetworkCommand = (
-)
+1. Return [=success=] with data [=CreateArrayFromList=](|deserialized key-value list|).
 
-</pre>
+</div>
 
-[=local end definition=]
+<div algorithm>
 
-<pre class="cddl local-cddl">
+Issue: TODO distinguish between `List` and `Array`.
 
-NetworkResult = (
-)
+To <dfn>deserialize value list</dfn> given |serialized value list|, |realm| and
+|session|:
 
-NetworkEvent = (
-    network.BeforeRequestSent //
-    network.FetchError //
-    network.ResponseStarted //
-    network.ResponseCompleted
-)
+1. Let |deserialized values| be a new list.
 
-</pre>
+1. For each |serialized value| in the |serialized value list|:
 
-A [=remote end=] has a <dfn>before request sent map</dfn> which is initially an
-empty map.  It's used to track the network events for which a
-<code>network.beforeRequestSent</code> event has already been sent.
+   1. Let |deserialized value| be result of [=trying=] to [=deserialize local value=]
+      given |serialized value|, |realm| and |session|.
 
-### Types ### {#module-network-types}
+   1. Append |deserialized value| to |deserialized values|;
 
-#### The network.BaseParameters type #### {#type-network-BaseParameters}
+1. Return [=success=] with data |deserialized values|.
 
-<pre class="cddl local-cddl">
-network.BaseParameters = {
-    context: BrowsingContext / null,
-    navigation: Navigation / null,
-    redirectCount: js-uint,
-    request: network.RequestData,
-    timestamp: js-uint,
-}
-</pre>
+</div>
 
-The <code>network.BaseParameters</code> type is an abstract type representing
-the data that's common to all network events.
+<div algorithm>
 
-Issue: Consider including the `sharedId` of the document node that initiated the
-request in addition to the context.
+To <dfn>deserialize local value</dfn> given |local protocol value|, |realm| and
+|session|:
 
-<div algorithm>
-To <dfn>get the base network event data</dfn> given |request| and |redirect count|:
+1. If |local protocol value| matches the [=script.RemoteReference=] production, return
+   [=deserialize remote reference=] of given |local protocol value|, |realm| and
+   |session|.
 
-1. Let |request data| be the result of [=get the request data=] with |request|.
+1. If |local protocol value| matches the [=script.PrimitiveProtocolValue=] production, return
+   [=deserialize primitive protocol value=] with |local protocol value|.
 
-<!-- TODO: update this to "[=request/navigation id=] once the fetch parts land-->
-1. Let |navigation| be |request|'s [=navigation id=].
+1. If |local protocol value| matches the <code>script.ChannelValue</code> production,
```

Comment:
Yeah, I noticed that too. I think the problem is that using `<dfn>type</dfn>` causes an error if we don't ever reference `type` elsewhere, which quite commonly happens if it's only used as part of the CDDL rather than referred to in prose. I guess in an ideal world Bikeshed would parse the CDDL and automatically create linkable references for us. In the meantime I think it would be possible to turn all the type definitions into actual linkable definitions, and suppress the error if we don't actually link to it, but that seems like something we should do in a different PR.

Reply:
yes, definitely not for this PR, thanks for the input!